### PR TITLE
New Rule: requireSpaceAfterComma

### LINKED ISSUE #1698: requireSpaceAfterBinaryOperators: ignores commas if set to true
After 9fc861494156a44b20616b710a5e9ffc5b328749 it's become very unwieldy to detect missing spaces after binary operators AND after commas in arrays. [All style guides use `requireSpaceAfterBinaryOperators: true`](https://github.com/jscs-dev/node-jscs/search?utf8=✓&q=requireSpaceAfterBinaryOperators+path%3Apresets&type=Code) and disapprove of a cramped style with no space after comma, but that option surprisingly won't detect it.

Examples below:

Code: `var a = [1,2, 3 +4];`
***
Rule: `{ "requireSpaceAfterBinaryOperators": [","] }`
```
Operator , should not stick to following expression at test.js :
     1 |var a = [1,2, 3 +4];
-------------------^

1 code style error found.
```
***
Rule: `{ "requireSpaceAfterBinaryOperators": true }`
```
Operator + should not stick to following expression at test.js :
     1 |var a = [1,2, 3 +4];
-------------------------^

1 code style error found.
```
***
Rule: `{ "requireSpaceAfterBinaryOperators": [ 
     "=", "+=", "-=", "*=", "/=", "%=", "<<=", ">>=", ">>>=",
     "&=", "|=", "^=",
     "+", "-", "*", "/", "%", "<<", ">>", ">>>", "&",
     "|", "^", "&&", "||", "===", "==", ">=",
     "<=", "<", ">", "!=", "!==",
     ","
] }`
```
Operator , should not stick to following expression at test.js :
     1 |var a = [1,2, 3 +4];
-------------------^

Operator + should not stick to following expression at test.js :
     1 |var a = [1,2, 3 +4];
-------------------------^


2 code style errors found.
```
2 errors is the expected result for `true`, but jscs produces only 1 error instead. I'd say it counts as a bug.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25925988-requirespaceafterbinaryoperators-ignores-commas-if-set-to-true?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by bjdixon**
New rule to match requireSpaceBeforeComma

Fixes #1698

### CODE REVIEW

**Comment by hzoo** [comment]
Nice looks pretty good - with extra tests for newlines/etc. We should probably update `disallowSpaceBeforeComma` too later (only has 2 tests)

File: lib/config/configuration.js [hunk 1]
```diff
@@ -1070,6 +1070,8 @@ Configuration.prototype.registerDefaultRules = function() {
 
     this.registerRule(require('../rules/require-space-before-comma'));
 
+    this.registerRule(require('../rules/require-space-after-comma'));
```
Comment (markelog) [inline lib/config/configuration.js]: Could remove extra newline there?

**Comment by markelog** [comment]
> We should probably update disallowSpaceBeforeComma too later (only has 2 tests)

Maybe, although if it has 100% coverage and no one complain about it...

Anyhow, it shouldn't be a concern of this pr :-)

File: lib/config/configuration.js (hunk 1 above)
Comment (bjdixon) [inline lib/config/configuration.js]: sorry I pushed before I saw this. I'll update the PR again.

File: grouping.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
     "disallowSemicolons",
     "requireSemicolons",
     "requireSpaceBeforeComma",
+    "requireSpaceAfterComma",
```
Comment (markelog) [inline grouping.json]: @mdevils be aware of this file :-)

**Comment by markelog** [comment]
Thank you for adding tests for the sister rule!

LGTM

**Comment by monk-time** [comment]
Great work, thanks!

**Pull request closed without merging**